File: src/logger.ts
// Sublimity pure RPC engine - Simply core implementation of pure RPC engine.
// Copyright (c) Kouji Matsui (@kekyo@mi.kekyo.net)
// License under MIT.

import { Logger } from "./types";

/**
 * Create a console logger.
 * @returns Logger object.
 */
export const createConsoleLogger = (): Logger => {
  return {
    debug: (message: string) => console.debug(message),
    info: (message: string) => console.info(message),
    warn: (message: string) => console.warn(message),
    error: (message: string) => console.error(message)
  };
};

File: dotnet/Ameba.Rpc/Logger.cs
// Ameba pure RPC engine - Simply core implementation of pure RPC engine.
// Copyright (c) Kouji Matsui (@kekyo@mi.kekyo.net)
// License under MIT.

using System;

namespace Ameba.Rpc;

/// <summary>
/// Console logger.
/// </summary>
public sealed class ConsoleLogger : ILogger
{
    public void Debug(string message) =>
        Console.Out.WriteLine(message);
    public void Information(string message) =>
        Console.Out.WriteLine(message);
    public void Warning(string message) =>
        Console.Error.WriteLine(message);
    public void Error(string message) =>
        Console.Error.WriteLine(message);
}
